### PR TITLE
Hive: Fix metadata file not found

### PR DESCRIPTION
**Pull request by lurnagao-dahua**
Hi, I get an Metadata not found in metadata location for table error when trying to query the data generated by flink

The reason is that in some cases, the e.getMessage() return null and it will throw NullPointerException, then skip checkCommitStatus, it may be delete metadataLocation, actually, metadata commit succeed.

It was introduced by [6570](https://github.com/apache/iceberg/pull/6570)

### CODE REVIEW

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java [hunk 1]
```diff
@@ -263,8 +263,8 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         throw e;
 
       } catch (Throwable e) {
-        if (e.getMessage()
-            .contains(
+        if (e.getMessage() != null
```
Comment (manuzhang) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java:266]: It'd better to create a util method for this pattern.

Comment (pvary) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java:266]: Where would we use this new method?
If we would use it only in this class, and only in 2 places, then I think it is better not to "hide" what we do behind a method somewhere else in the code.

**Review by pvary — APPROVED** [review]

**Review by pvary — COMMENTED** [review]
Oh... accidentally chose "Approve".

While I think the PR is good, I wanted to wait @manuzhang's review before moving forward.

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java (hunk 1 above)
Comment (manuzhang) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java:266]: If we had the method in the first place, I don't think we would miss the null check here. With this method, we can ensure the null check will be applied in any other places.

**Comment by manuzhang** [comment]
@lurnagao-dahua please check styles. 

> The reason is that in some cases, the e.getMessage() return null and it will throw NullPointerException, then skip checkCommitStatus, it may be delete metadataLocation, actually, metadata commit succeed.

Is it possible to add a UT for this case?

**Comment by lurnagao-dahua** [comment]
> @lurnagao-dahua请检查样式。
> 
> > 原因是在某些情况下，e.getMessage()返回null，会抛出NullPointerException，然后跳过checkCommitStatus，可能是删除metadataLocation，实际上元数据提交成功。
> 
> 是否可以为这种情况添加 UT？

**Comment by lurnagao-dahua** [comment]
> @lurnagao-dahua please check styles.
> 
> > The reason is that in some cases, the e.getMessage() return null and it will throw NullPointerException, then skip checkCommitStatus, it may be delete metadataLocation, actually, metadata commit succeed.
> 
> Is it possible to add a UT for this case?

Hi, I found the original pr similar to this one [701](https://github.com/apache/iceberg/pull/701), then I will try to write an UT

**Review by nastra — APPROVED** [review]
LGTM, it seems we're using the same pattern in a few other places, but having a test to reproduce the underlying issue would be great

**Comment by lurnagao-dahua** [comment]
Thank you for your response!
In my case flink streaming write to iceberg
1.Hive metastore has been full GC  continuously so it will throw SocketTimeoutException: Read timed out(`hive.metastore.client.socket.timeout` default 600s)
2.hiveTableOperations commit thread call `Thread.sleep(retryDelaySeconds * 1000)` to retry
3.The Flink checkpoint timeout time is less than 600s and Interrupt it, then throw InterruptedException and not message

I have been thinking for a while and I have some doubts about this UT.  Can you give me some advice?

**Comment by pvary** [comment]
@lurnagao-dahua: If you check https://github.com/apache/iceberg/blob/3caa3a28d07a2d08b9a0e4196634126f1e016d6a/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java, you can find plenty of examples for commit errors. Maybe if we could do something similar, like throwing an exception without a message. It would be nice to have a test.

OTOH, if the test is more than 50 lines, it would cost us more in the upkeep of the test in the long run, than what we gain with testing a null check. In this case I would skip addig the extra code, following the example of #701.

**Comment by lurnagao-dahua** [comment]
> @lurnagao-dahua: If you check https://github.com/apache/iceberg/blob/3caa3a28d07a2d08b9a0e4196634126f1e016d6a/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java, you can find plenty of examples for commit errors. Maybe if we could do something similar, like throwing an exception without a message. It would be nice to have a test.
> 
> OTOH, if the test is more than 50 lines, it would cost us more in the upkeep of the test in the long run, than what we gain with testing a null check. In this case I would skip addig the extra code, following the example of #701.

Hi,Thank you for your suggestion
I added a unit test.It will throw runtimeException without message and result in throw CommitStateUnknownException,the meesage will be null + COMMON_INFO:
```
public CommitStateUnknownException(Throwable cause) {
  super(cause.getMessage() + "\n" + COMMON_INFO, cause);
}
```

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java [hunk 2]
```diff
@@ -397,6 +397,30 @@ public void testLockExceptionUnknownSuccessCommit() throws TException, Interrupt
         .isTrue();
   }
 
+  @Test
+  public void testCommitExceptionWithoutMessage() throws TException, InterruptedException {
```
Comment (nastra) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java:401]: can be simplified to
```
@Test
  public void testCommitExceptionWithoutMessage() throws TException, InterruptedException {
    Table table = catalog.loadTable(TABLE_IDENTIFIER);
    HiveTableOperations ops = (HiveTableOperations) ((HasTableOperations) table).operations();

    TableMetadata metadataV1 = ops.current();
    table.updateSchema().addColumn("n", Types.IntegerType.get()).commit();

    ops.refresh();

    HiveTableOperations spyOps = spy(ops);
    doThrow(new RuntimeException()).when(spyOps).persistTable(any(), anyBoolean(), any());

    assertThatThrownBy(() -> spyOps.commit(ops.current(), metadataV1))
        .isInstanceOf(CommitStateUnknownException.class)
        .hasMessageStartingWith("null\nCannot determine whether the commit was successful or not");
  }
```

Comment (lurnagao-dahua) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java:401]: > @Test
>   public void testCommitExceptionWithoutMessage() throws TException, InterruptedException {
>     Table table = catalog.loadTable(TABLE_IDENTIFIER);
>     HiveTableOperations ops = (HiveTableOperations) ((HasTableOperations) table).operations();
> 
>     TableMetadata metadataV1 = ops.current();
>     table.updateSchema().addColumn("n", Types.IntegerType.get()).commit();
> 
>     ops.refresh();
> 
>     HiveTableOperations spyOps = spy(ops);
>     doThrow(new RuntimeException()).when(spyOps).persistTable(any(), anyBoolean(), any());
> 
>     assertThatThrownBy(() -> spyOps.commit(ops.current(), metadataV1))
>         .isInstanceOf(CommitStateUnknownException.class)
>         .hasMessageStartingWith("null\nCannot determine whether the commit was successful or not");
>   }

Thank you very much, I have simplified unit test.

**Review by nastra — APPROVED** [review]

**Pull request merged**